src/Impl/Utils/MultipleJsonsSplitter.php

```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types=1);
+
+namespace Streamx\Clients\Ingestion\Impl\Utils;
+
+class MultipleJsonsSplitter {
+
+    private const ESCAPED_QUOTE = '\"';
+    private const ESCAPED_QUOTE_TEMPORARY_REPLACEMENT = '^kj-N2hpS.';
+
+    /**
+     * @param $jsonsString string containing one or more JSON strings. They can be separated with any or no separator
+     * @return array of
+     */
+    static function splitToSingleJsons(string $jsonsString): array {
+        $resultJsons = [];
+        $openedBracesCounter = 0;
+        $currentJsonStartIndex = 0;
+        $insideQuotes = false;
+
+        $escapedJsonsString = self::maskEscapedQuotes($jsonsString);
+        $stringLength = strlen($escapedJsonsString);
+
+        for ($i = 0; $i < $stringLength; $i++) {
+            $char = $escapedJsonsString[$i];
+            if ($char == '"') {
+                $insideQuotes = !$insideQuotes;
+            }
+            if ($insideQuotes) {
+                continue;
+            }
```

Comment:
Added the two above if's.

If we are inside json field name or value (inside " ") - then skip searching for opening { and closing } tags